include/messages.h

```diff
@@ -223,12 +231,17 @@ enum struct ContentType : uint8_t
   commit = 3,
 };
 
-struct Proposal : public tls::variant<ProposalType, Add, Update, Remove>
+struct Proposal
 {
-  using parent = tls::variant<ProposalType, Add, Update, Remove>;
-  using parent::parent;
+  std::variant<Add, Update, Remove> content;
+  TLS_SERIALIZABLE(content);
+  TLS_TRAITS(tls::variant<ProposalType>);
+};
 
-  static const ContentType type;
+struct ProposalID {
+  bytes id;
+  TLS_SERIALIZABLE(id);
+  TLS_TRAITS(tls::vector<1>);
 };
 
```

Comment:
do we need both variations here .. Serializable and Traits ? wouldn't that end in double serialization of the id

Reply:
or we need TLS_TRAITS{tls::pass)